src/torrent.cpp

```diff
@@ -3228,7 +3238,13 @@ bool is_downloading_state(int const st)
 				if (!aep->start_sent && r.event == tracker_request::started)
 					aep->start_sent = true;
 				if (!aep->complete_sent && r.event == tracker_request::completed)
+				{
 					aep->complete_sent = true;
+					// we successfully reported event=completed to one tracker. Don't
+					// send it to any other ones from now on (there may be other
+					// announces outstanding right now though)
+					m_complete_sent = true;
```

Comment:
this is a material change, setting the torrent-wide `m_complete_sent` when *any* tracker successfully announces with `event=completed`